### PR TITLE
Container locator

### PR DESCRIPTION
**Pull request by NigelGreenway**
Initial container locator plugin code

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
+  "name":        "league/tactician-container",
+  "description": "Tactician integration with the PHP League DI container",
+  "keywords":    ["tactician", "di", "container", "league"],
+  "license":     "MIT",
+  "authors": [
+    {
+      "name": "Ross Tuck",
+      "homepage": "http://tactician.thephpleague.com"
+    }
```
Comment (rosstuck) [inline composer.json:10]: Add yourself as first author here, man. :smile:

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  ],
+  "require": {
+    "php":  ">=5.5",
+    "league/tactician": "~0.3",
```
Comment (rosstuck) [inline composer.json]: Can you change this constraint to ^0.3 instead of ~0.3?

File: src/ContainerLocator.php [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+     * @param Container $container                The Container object
+     * @param array     $commandClassToHandlerMap The Command/Handler mapping
+     */
+    public function __construct(
+        Container $container,
+        array $commandClassToHandlerMap = []
+    ) {
+        $this->container = $container;
+
+        if (empty($commandClassToHandlerMap) === false) {
```
Comment (rosstuck) [inline src/ContainerLocator.php]: You can drop the empty check here. If it's not an array, the typehint will catch it. If it is an array, the addHandlers won't loop (and should always properly handle an empty array anyways)

File: src/ContainerLocator.php [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    /**
+     * Retrieve handler for the given command
+     *
+     * @param Command $command The command object
+     *
+     * @return Object
+     *
+     * @throws MissingHandlerException
```
Comment (rosstuck) [inline src/ContainerLocator.php:81]: Some of the comments have lines between annotations, others don't. Could you make these consistent by removing the lines?

File: src/Exception/MissingCommandException.php [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace League\Tactician\Container\Exception;
+
+use League\Tactician\Command;
+use League\Tactician\Exception\Exception;
+
+/**
+ * Command has not been added to the mapping array.
+ */
+class MissingCommandException extends \OutOfBoundsException implements Exception
```
Comment (rosstuck) [inline src/Exception/MissingCommandException.php]: I'd recommend removing this custom exception and throw the `MissingHandlerException` in Tactician core instead. That would let folks always typehint for one particular exception, regardless of their background configuration. Also, there's a semantic error here in that we're missing a Handler, not a Command. :smile:

**Comment by rosstuck** [comment]
Added a couple comments. If you can resolve these, I see no reason to not merge. :+1:

**Comment by NigelGreenway** [comment]
Hi @rosstuck,

Done the changes as suggested.

Hope thats better ;)

**Comment by rosstuck** [comment]
Looking good! Thanks a ton for all your work and patience on this. :smile: 

I'm gonna merge ASAP but not tag a release yet, since I'm running around and sending PRs to all the different repos in preparation for the command interface being removed ( https://github.com/thephpleague/tactician/issues/43 ). I'll do that later today and wait for your feedback on that PR before merging as well.

Once again, many thanks! :smile:

**Pull request merged**